### ISSUE TITLE
satellite overlay objects

### ISSUE THREAD

**Issue by bjorn-stevens** [orcestra-campaign/pyorcestra#39]
For overlaying goes data, and perhaps other satellites, there are some helper functions, which I think should be part of pyorcestra.   For instance I use the following from Divya:

```python
from goes2go.data import goes_nearesttime, goes_latest
import xarray as xr
def _create_GOES_variable(
        goes_object: xr.Dataset, variable: str, gamma:float = 1.2
        ):
    GOES_PRODUCT_DICT = {
        "AirMass": goes_object.rgb.AirMass,
        "AirMassTropical": goes_object.rgb.AirMassTropical,
        "AirMassTropicalPac": goes_object.rgb.AirMassTropicalPac,
        "Ash": goes_object.rgb.Ash,
        "DayCloudConvection": goes_object.rgb.DayCloudConvection,
        "DayCloudPhase": goes_object.rgb.DayCloudPhase,
        "DayConvection": goes_object.rgb.DayConvection,
        "DayLandCloud": goes_object.rgb.DayLandCloud,
        "DayLandCloudFire": goes_object.rgb.DayLandCloudFire,
        "DaySnowFog": goes_object.rgb.DaySnowFog,
        "DifferentialWaterVapor": goes_object.rgb.DifferentialWaterVapor,
        "Dust": goes_object.rgb.Dust,
        "FireTemperature": goes_object.rgb.FireTemperature,
        "NaturalColor": goes_object.rgb.NaturalColor(gamma=gamma),
        "NightFogDifference": goes_object.rgb.NightFogDifference,
        "NighttimeMicrophysics": goes_object.rgb.NighttimeMicrophysics,
        "NormalizedBurnRatio": goes_object.rgb.NormalizedBurnRatio,
        "RocketPlume": goes_object.rgb.RocketPlume,
        "SeaSpray": goes_object.rgb.SeaSpray(gamma=gamma),
        "SplitWindowDifference": goes_object.rgb.SplitWindowDifference,
        "SulfurDioxide": goes_object.rgb.SulfurDioxide,
        "TrueColor": goes_object.rgb.TrueColor(gamma=gamma),
        "WaterVapor": goes_object.rgb.WaterVapor,
    }
    return GOES_PRODUCT_DICT[variable]
```
    
 With this it is pretty easy to get GOES data and overlay it.. I just thought that the above is a bit more general purpose and perhaps should be added to the pyorcestra tools, but did not have an obvious suggestion as to where.  

With this definition to overlay a BAHAMAS flight track on GOES is just a matter of the following
 
 ```python
from orcestra.flightplan import  LatLon, path_preview
import matplotlib.pyplot as plt
import seaborn as sns
import cartopy.crs as ccrs
sns.set_context("talk")

flight_name = 'HALO-20240811a'
flight_date = flight_name[5:9]+'-'+flight_name[9:11]+'-'+flight_name[11:13]
g2g_0811_16utc = goes_nearesttime(flight_date+'T16:00', satellite=16, product="ABI", domain = "F")
goes_variable  = _create_GOES_variable(g2g_0811_16utc, "TrueColor")
tracks = xr.open_dataset('/Volumes/ORCESTRA/'+flight_name+'/bahamas/QL_'+flight_name+'_BAHAMAS_V01.nc')
path   = LatLon(lat=tracks['IRS_LAT'], lon=tracks['IRS_LON'], label=flight_name)
f
ig, ax = plt.subplots(
    figsize=(15, 8),
    facecolor='white',
    subplot_kw  ={"projection": ccrs.PlateCarree()}
)
path_preview(path,ax=ax,show_waypoints=False,color='#FF5349')
ax.imshow(goes_variable, transform=g2g_0811_16utc.rgb.crs, regrid_shape=3500,
              interpolation='nearest') 
fig.savefig(flight_name+'-track.png',bbox_inches='tight')
 ```

**Comment by lkluft** [comment]
Yes, I think this feature would be a nice addition!

At the moment we have `orcestra.sat` which mainly deals with satellite **tracks** and `orcestra.weathermaps` which so far only plots satellite images... I see where your confusion comes from 😬

For now, I would suggest putting it next to the MODIS function in `orcestra.weathermaps`, but expect the location to change in the future.

**Comment by lkluft** [comment]
Closed in #43
